rts/Sim/Weapons/Weapon.cpp

```diff
@@ -1268,12 +1271,57 @@ float3 CWeapon::GetUnitLeadTargetPos(const CUnit* unit) const
 	return aimPos;
 }
 
+// debug print includes
+//#include<iostream>
+//#include<string>
 
 float3 CWeapon::GetLeadVec(const CUnit* unit) const
 {
-	const float predictTime = GetPredictedImpactTime(unit->pos);
+	float predictTime = GetPredictedImpactTime(unit->pos);
 	const float predictMult = mix(predictSpeedMod, 1.0f, weaponDef->predictBoost);
 
+	if (accurateLeading == true) {
+		if (weaponDef->projectileType == WEAPON_EXPLOSIVE_PROJECTILE) {
+			const float gravity = mix(mapInfo->map.gravity, -weaponDef->myGravity, weaponDef->myGravity != 0.0f);
+			//std::cout << "weapon gravity = " << gravity << std::endl;
+
+			if (gravity < 0) {
+				// precise target leading
+				// newton iterations are too unstable, due to impossible to intercept targets
+				// use fixed point iterations
+				// At most 32 are allowed, but mostly just 2 iterations are needed, and the most I have seen is 10
+				float3 dist = unit->pos - weaponMuzzlePos;
+
+				const float ee = 0.25f * (gravity) * (gravity);
+				float t1 = 1.0f;
+				float temp1 = 1.0f;
+				float temp2 = 1.0f;
+				float cc = 1.0f;
+				for (int ii = 0; ii < 32; ii++) {
```

Comment:
Perhaps `accurateLeading` should be a number, and then
```suggestion
				for (int ii = 0; ii < accurateLeading; ii++) {
```

Reply:
A future generation of readers will appreciate if you could outline the math details of iteration method you've implemented or you can leave a link to some reliable source, like wikipedia.

Reply:
Reviewed and further elaborated on the iteration method